tests/virt_autotest/alp_guest_installation.pm

```diff
@@ -1,66 +0,0 @@
-# Copyright 2023 SUSE LLC
-# SPDX-License-Identifier: FSFAP
-#
-# Summary: This file handles the guest installation for ALP.
-#          It supports configuring vm profiles with template.
-#
-# Maintainer: Alice <xlai@suse.com>, or VT squad <qe-virt@suse.de>
-
-package alp_guest_installation;
-
-use base unified_guest_installation;
-use strict;
-use warnings;
-use testapi;
-use Data::Dumper;
-
-sub edit_guest_profile_with_template {
```

Comment:
I think this is a nice-to-have feature to allow configuring vm profiles with template, which can avoid so many similar config files in guest_params_xml_files with only a few differences. It is still a valid nice to have for those slem_6_xx, I think.

Reply:
Definitely, this code piece can't be kept without changes because of the redesign of base module. I mean the idea is still applicable, because there are still so many xml files which have great similarity but limited changes. So using template can be a way to solve that.

Reply:
It is not appealing to me, because it will lead to parameters explosion at the test suite level and also at every place that need them or display them. Using profile hide such complexity and have better isolation. Using separate profile is convenient for me. It can help avoid chaos and also have flexibility.

Reply:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/18488#discussion_r1501962567 fits here too.

It can be subjective whether some feature is appealing to someone. The point is that it is already a feature in the original code to allow configuring vm xml from template. So anyone would like that they can use it as a feature, and for those do not like it, they can continue original way with independent xml file. It is up to the code consumers how they choose. However with changes in this PR, it will fully break the function. But we never had discussions and agreements to discard it, nor it is required in the ticket scope.

Reply:
I keep the alp lib file. But current alp code in these modules have to be removed, because:
1. They are outdated already. For example, it still refers to "openSUSE" when generates junit.
2. They are not clean and not conform to usage of parameters in base base. So they are not useful anymore.
The alp lib file is you largest asset  which is kept. But the alp code must be cleaned up and added back (if necessary) in better form and format.

Reply:
By the way, guest autoyast file already uses template, but it is not very suitable for guest profile.

Reply:
> I keep the alp lib file. But current alp code in these modules have to be removed, because:
> 
> 2. They are not clean and not conform to usage of parameters in base base. So they are not useful anymore.

That's the point. It should be fixed then, rather than deleted.

Reply:
It would be better to clean them up in this refactor. 
1. Outdated and not good enough code really exhibits "not good" example and also make overall structure looks "not unified and consolidated".  
2. And It may also detract people from the main thing and more proper way to improve/enhance further in the future. 
3. If there is new requirement, corresponding functionality will be added back with better form, nothing will be forgotten. 
There is no need to let the "outdated and not useful" thing linger too long time, otherwise there will be no opportunity for the better. This is also the benefit that "refactoring" can bring at some point. Sometime one step forward is not easy, but it still needs to take some day by someone.

Reply:
I think the designer should consider uses' requirements, rather than focusing the tidy codes only. The users know what they need and what they like, and there has already supporting codes, it is not a good choice to drop existing codes.

Reply:
I think the designer should consider uses' requirements, rather than focusing the tidy codes only. The users know what they need and what they like, and there has already supporting codes, it is not a good choice to drop existing codes.

Reply:
1. I do not removing supporting code for the sake of tidying up code, which is not the way in which I do things. I cleaned them up because they are already outdated and do not serve effective purpose anymore. For example, current existing code still refers to "entering/exiting a kvm container" which is already totally obsolete. What really does guest installation is the original base module, so I have to clean up. 

2. I do not only take user requirement into consideration, but also user experience, so user can install guest using either pre-built image (raw/qcow/encrypted) or HTTP repo conveniently, smoothly and stably. So this also involves overall structure, but as designer I do not implement user requirement in unpleasant way.

3. As far as existing codes or the action to clean them up is concerned:
    * They are already outdated and do not serve effective purpose anymore, which means they do not play a role in installing guest anymore.
    * I ever mentioned that ```If there is new requirement, corresponding functionality will be added back with better form, nothing will be forgotten.``` which does not mean existing code can even partially support to some extent, what I meant is new requirement can be implemented in better way instead of inheriting "outdated and not good example" which distracts people from more proper way. So clean-up has to be done. Please refer to [this comment](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/18488#discussion_r1502315656) again.
    * There is no missing functionality (but add support for pre-built image) compared with before doing clean-up. Clean-up here is to reorganizing/remving/improving code without affecting its original functionality.

4. I think you need to know more about "Refactoring" and "Technical Debt" which perfectly applies to this case:
   * Refactoring in software engineering is reorganizing code without affecting its original functionality. Refactoring aims to improve internal code by making modest changes that do not affect the code's exterior behavior. Computer programmers and software developers refactor code to improve software design, structure, and implementation. As a result, Refactoring increases code readability while decreasing complications. Refactoring can also assist software engineers in locating faults or vulnerabilities in their code.
   * Ward Cunningham proposed the notion of "technical debt" about the unclean code.
   * Here is the [full article](https://www.scaler.com/topics/software-engineering/refactoring-in-software-engineering/).